nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/MapRecord.java

```diff
@@ -312,7 +312,50 @@ public String toString() {
 
     @Override
     public Map<String, Object> toMap() {
-        return Collections.unmodifiableMap(values);
+        return toMap(false);
```

Comment:
This retains current behavior but is that what we want? Or is it a bug that should be fixed for the same reason you added the overloaded method?